### ISSUE TITLE
Change directory within custom command?

### ISSUE THREAD

**Issue by Hipfox** [nushell/nushell#5218]
### Describe the bug

can't change directory after excute custom command.

### How to reproduce

```
/home/harry〉def cdtest [] {
:::   let path = '/tmp'
:::   echo $path
:::   cd $path
::: }
/home/harry〉cdtest
/tmp
/home/harry〉
```

the result not work with cd /tmp

### Expected behavior

I expected nu to change directory after execute def command.

### Configuration

/home/harry〉version | transpose key value | to md --pretty
| key                | value                                     |
| ------------------ | ----------------------------------------- |
| version            | 0.61.0                                    |
| branch             |                                           |
| tag                |                                           |
| short_commit       | 3783c19d                                  |
| commit_hash        | 3783c19d02f4a3b2f35cd26b79daabcb70b04880  |
| commit_date        | 2022-04-12 20:38:15 +00:00                |
| build_os           | linux-x86_64                              |
| rust_version       | rustc 1.60.0 (Arch Linux rust 1:1.60.0-1) |
| cargo_version      | cargo 1.60.0                              |
| pkg_version        | 0.61.0                                    |
| build_time         | 2022-04-12 23:20:08 +00:00                |
| build_rust_channel | release                                   |
| features           | dataframe, default, trash, which, zip     |
| installed_plugins  |                                           |

**Comment by fdncred** [comment]
This is by design. Please see the book https://www.nushell.sh/book/thinking_in_nushell.html#nushell-s-environment-is-scoped

**Comment by jiripospisil** [comment]
New URL: https://www.nushell.sh/book/thinking_in_nu.html#nushell-s-environment-is-scoped

tl;dr: Use `def-env` instead of `def`.
